homeassistant/components/homewizard/config_flow.py

```diff
@@ -166,6 +168,28 @@ async def async_step_discovery_confirm(
             },
         )
 
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Handle re-auth if API was disabled."""
+
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm reauth dialog."""
+
+        if user_input is not None:
+            assert self.entry is not None
+
+            await self._async_try_connect_and_fetch(self.entry.data[CONF_IP_ADDRESS])
```

Comment:
I see that this method raises an Abort. But given that you know the device exists and you just need them to enable the API, it would make more sense to just report an error. That way the user can press SUBMIT and re-try the authentication.

Reply:
Not considered, this is _way_ better. Thanks! I've adjusted the config flow to show an error when appropriate, instead of aborting. 

I only having a hard time testing this for discovery in the dev container. The unit tests tell me it works but I can't check it in real use. Any ideas?

Reply:
If you run your dev container with `—net=host`, it will attach itself to the host network and work with zeroconf.